packages/astro-embed-baseline-status/BaselineStatus.astro

```diff
@@ -0,0 +1,293 @@
+	: {
+			baseline: {
+				status: 'no_data',
+			},
+			name: Astro.props.id || 'Unknown feature',
+	  };
+
+const baseline = feature.baseline.status || 'no_data';
+
+const title = BASELINE_DEFS[baseline].title;
+const baselineDate = getBaselineDate(feature);
+const description = getDescriptionDate(baseline, baselineDate);
+const year =
+	baseline === 'newly' && baselineDate ? baselineDate.split(' ')[1]! : '';
+const { chrome, edge, firefox, safari } = feature.browser_implementations || {};
+
+function getAriaLabel(
+	title: string,
+	year: string,
+	newlyAvailable: boolean,
+	chrome = 'no',
+	edge = 'no',
+	firefox = 'no',
+	safari = 'no'
+) {
+	if (title === 'Unknown availability') {
+		chrome = edge = firefox = safari = 'unknown';
+	}
+	return `Baseline: ${title}${year ? ` ${year}` : ''}${
+		newlyAvailable ? ` (newly available)` : ''
+	}. Supported in Chrome: ${
+		chrome === 'available' ? 'yes' : chrome
+	}. Supported in Edge: ${
+		edge === 'available' ? 'yes' : edge
+	}. Supported in Firefox: ${
+		firefox === 'available' ? 'yes' : firefox
+	}. Supported in Safari: ${safari === 'available' ? 'yes' : safari}.`;
+}
+---
+
+<baseline-status>
+	<h1>{feature.name}</h1>
+	<details>
+		<summary
+			aria-label={getAriaLabel(
+				title,
+				year,
+				baseline === 'newly',
+				chrome?.status,
+				edge?.status,
+				firefox?.status,
+				safari?.status
+			)}
+		>
+			<BaselineIcon support={baseline} />
+			<div class="baseline-status-title" aria-hidden="true">
+				<div>
+					{
+						!(baseline === 'limited' || baseline === 'no_data') && (
+							<strong>Baseline</strong>
+						)
+					}
+					{title}
+					{year}
+					{
+						baseline === 'newly' && (
+							<span class="baseline-badge">newly available</span>
+						)
+					}
+				</div>
+				<div class="baseline-status-browsers">
+					<span>
+						<BrowserIcon browser="chrome" />
+						<SupportIcon {baseline} browserImplementation={chrome} />
+					</span>
+					<span>
+						<BrowserIcon browser="edge" />
+						<SupportIcon {baseline} browserImplementation={edge} />
+					</span>
+					<span>
+						<BrowserIcon browser="firefox" />
+						<SupportIcon {baseline} browserImplementation={firefox} />
+					</span>
+					<span>
+						<BrowserIcon browser="safari" />
+						<SupportIcon {baseline} browserImplementation={safari} />
+					</span>
+				</div>
+			</div>
+			<span class="open-icon" aria-hidden="true">
+				<svg
+					xmlns="http://www.w3.org/2000/svg"
+					width="11"
+					height="7"
+					viewBox="0 0 11 7"
+					fill="none"
+				>
+					<path
+						d="M5.5 6.45356L0.25 1.20356L1.19063 0.262939L5.5 4.59419L9.80937 0.284814L10.75 1.22544L5.5 6.45356Z"
+						fill="currentColor"></path>
+				</svg>
+			</span>
+		</summary>
+		<p set:html={description} />
+		<p
+			set:html={baseline === 'no_data'
+				? ''
+				: `<a href="https://github.com/web-platform-dx/web-features/blob/main/features/${feature.feature_id}.yml" target="_top">Learn more</a>`}
```

Comment:
I've been trying to figure out if there's a way to get a more useful link (i.e. MDN or CanIUse). The GitHub link is unhelpful 😅

Reply:
Yeah… the API does return an array of spec links (where available I assume), see https://api.webstatus.dev/v1/features/has for example which includes:

```json
{ 
	"spec": {
		"links": [{"link":"https://drafts.csswg.org/selectors-4/#relational"}]
	}
}
```

Although that’s still not _super_ helpful in many cases.

Another option could be linking to `webstatus.dev` instead of GitHub. It uses the same feature IDs so should work predictably, e.g. https://webstatus.dev/features/has.

Reply:
Updated the links to point to the relevant webstatus.dev page. Seems more useful than the random YAML file on GitHub. (Also updated so the links include the feature name, not just the nondescript “Learn more”.)